### PR TITLE
Updated utility device types according to specs

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Utility device types was not as per specs.

#### Change overview
Updated utility device types according to specs

#### Testing
How was this tested? (at least one bullet point required)
* Verified that the xml is in sync with the specification.

### CODE REVIEW

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
There are a bunch of differences here wrt spec (_after_ the changes for this PR).  Please create one PR per device type, so they can be reviewed independently.

**Review by woody-apple — APPROVED** [review]
For future changes, please keep them separated by PR.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -144,50 +124,30 @@ limitations under the License.
...
                 <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
                 <requireAttribute>SERVER_LIST</requireAttribute>
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="true" server="false" clientLocked="false" serverLocked="true">
-                <requireAttribute>BINDING</requireAttribute>
-            </include>
             <include cluster="OTA Software Update Provider" client="false" server="true" clientLocked="true" serverLocked="true"></include>
             <include cluster="OTA Software Update Requestor" client="true" server="false" clientLocked="true" serverLocked="true"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:134]:
```suggestion
            <include cluster="OTA Software Update Requestor" client="true" server="false" clientLocked="false" serverLocked="true"></include>
```
This is optional in the spec.

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -144,50 +124,30 @@ limitations under the License.
...
                 <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
                 <requireAttribute>SERVER_LIST</requireAttribute>
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="true" server="false" clientLocked="false" serverLocked="true">
-                <requireAttribute>BINDING</requireAttribute>
-            </include>
-            <include cluster="Bridged Actions" client="false" server="true" clientLocked="true" serverLocked="true"></include>
+            <include cluster="Bridged Actions" client="false" server="false" clientLocked="true" serverLocked="false"></include>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:150]:
```suggestion
            <include cluster="Actions" client="false" server="false" clientLocked="true" serverLocked="false"></include>
```

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -197,20 +157,13 @@ limitations under the License.
...
-                <requireCommand>TriggerEffect</requireCommand>
-            </include>
             <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
                 <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
                 <requireAttribute>SERVER_LIST</requireAttribute>
                 <requireAttribute>CLIENT_LIST</requireAttribute>
                 <requireAttribute>PARTS_LIST</requireAttribute>
             </include>
-            <include cluster="Binding" client="true" server="false" clientLocked="false" serverLocked="true">
+            <include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="true">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:166]: Should just be removed, no?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 4]
```diff
@@ -197,20 +157,13 @@ limitations under the License.
...
                 <requireAttribute>BINDING</requireAttribute>
             </include>
             <include cluster="Bridged Device Basic" client="false" server="true" clientLocked="true" serverLocked="true">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:169]: Power Source and Power Source Configuration are both optional.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
Like I said, there are a bunch of differences wrt spec here.  Please check that this is aligned with the spec.

**Comment by jadhavrohit924** [comment]
Will add per device type separate PR soon to avoid confusion and easy to review. Closing this.

**Pull request closed without merging**